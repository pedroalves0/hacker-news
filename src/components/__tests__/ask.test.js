import React from 'react';
import { shallow } from 'enzyme';
import Ask from '../ask';

describe('components/ask', () => {
    test('should render correctly', () => {
        const wrapper = shallow(
            <Ask
                by={'author'}
                text={'text'}
                time={1512157581}
                title={'title'}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render the ask text if the toggle button is clicked', () => {
        const wrapper = shallow(
            <Ask
                by={'author'}
                text={'text'}
                time={1512157581}
                title={'title'}
            />
        );

        expect(wrapper).toMatchSnapshot();

        wrapper.find('.ask-title-toggle').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });

    test('should not render the toggle button if theres no text', () => {
        const wrapper = shallow(
            <Ask
                by={'author'}
                time={1512157581}
                title={'title'}
            />
        );

        expect(wrapper.find('.ask-title-toggle').exists()).toBe(false);
    });
});