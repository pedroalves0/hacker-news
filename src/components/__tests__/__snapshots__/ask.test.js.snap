// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ask should render correctly 1`] = `
<div
  className="ask"
>
  <div
    className="ask-title"
  >
    <span>
      title
    </span>
    <button
      className="ask-title-toggle"
      onClick={[Function]}
    >
      <FaPlusSquare />
    </button>
  </div>
  <div
    className="ask-info"
  >
    <span>
      by 
      <span
        className="t-ital"
      >
        author
         - 
      </span>
    </span>
    <TimeAgo
      component="time"
      date={2017-12-01T19:46:21.000Z}
      formatter={[Function]}
      live={true}
      maxPeriod={Infinity}
      minPeriod={0}
      now={[Function]}
    />
  </div>
</div>
`;

exports[`components/ask should render the ask text if the toggle button is clicked 1`] = `
<div
  className="ask"
>
  <div
    className="ask-title"
  >
    <span>
      title
    </span>
    <button
      className="ask-title-toggle"
      onClick={[Function]}
    >
      <FaPlusSquare />
    </button>
  </div>
  <div
    className="ask-info"
  >
    <span>
      by 
      <span
        className="t-ital"
      >
        author
         - 
      </span>
    </span>
    <TimeAgo
      component="time"
      date={2017-12-01T19:46:21.000Z}
      formatter={[Function]}
      live={true}
      maxPeriod={Infinity}
      minPeriod={0}
      now={[Function]}
    />
  </div>
</div>
`;

exports[`components/ask should render the ask text if the toggle button is clicked 2`] = `
<div
  className="ask"
>
  <div
    className="ask-title"
  >
    <span>
      title
    </span>
    <button
      className="ask-title-toggle"
      onClick={[Function]}
    >
      <FaMinusSquare />
    </button>
  </div>
  <div
    className="ask-info"
  >
    <span>
      by 
      <span
        className="t-ital"
      >
        author
         - 
      </span>
    </span>
    <TimeAgo
      component="time"
      date={2017-12-01T19:46:21.000Z}
      formatter={[Function]}
      live={true}
      maxPeriod={Infinity}
      minPeriod={0}
      now={[Function]}
    />
  </div>
  <p
    className="ask-content"
    dangerouslySetInnerHTML={
      Object {
        "__html": "text",
      }
    }
  />
</div>
`;
